pkg/otel/common/arrow/attributes.go

```diff
@@ -236,10 +236,14 @@ func (c *Attributes16Accumulator) AppendWithID(parentID uint16, attrs pcommon.Ma
 	}
 
 	attrs.Range(func(key string, v pcommon.Value) bool {
+		// Attribute without key or value are ignored.
 		if key == "" {
 			// Skip entries with empty keys
 			return true
 		}
+		if v.Type() == pcommon.ValueTypeEmpty {
```

Comment:
Should other append functions in this file that skip empty keys follow this convention?

Reply:
Also is there a meaningful difference between `v.Type() == pcommon.ValueTypeEmpty` and `v.Type() == pcommon.ValueTypeStr && v.Str() == ""`

Reply:
First question: Yes good catch

Second question: Empty strings are actually detected and encoded in Arrow. So empty string won't trigger the same bug.